### PR TITLE
Add Gradle task that makes LLVM dependency smaller

### PR DESCRIPTION
**Pull request by sbogolepov**
Whitelists and blacklists are not tuned yet and (probably) will be updated in the next PR.

### CODE REVIEW

**Comment by msink** [comment]
on mingw: windres does not work without gcc

**Comment by sbogolepov** [comment]
Thanks. This PR is laying ground for the future work, so these lists could be wrong :)

**Comment by msink** [comment]
Missing .dll:
```
--- c-index-test.exe 
libclang.dll
libLLVM.dll
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
libffi-6.dll
libz3.dll
libxml2-2.dll
libiconv-2.dll
liblzma-5.dll
 
--- clang.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
libz3.dll
 
--- clang++.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
libz3.dll
 
--- ld.lld.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- lld.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- llvm-ar.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- llvm-config.exe 
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- llvm-cov.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- llvm-dis.exe 
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- llvm-link.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- llvm-profdata.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
 
--- wasm-ld.exe 
zlib1.dll
libwinpthread-1.dll
libgcc_s_seh-1.dll
libstdc++-6.dll
```

**Comment by sbogolepov** [comment]
Thank you!

**Comment by msink** [comment]
.dll dependensies were listed by https://github.com/msink/win32-dlltree https://github.com/JetBrains/kotlin-native/issues/1981#issuecomment-449958905

File: dependencyPacker/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    tasks.create(downloadTaskName) {
+        doFirst {
+            propertiesLoader.downloadDependencies()
+        }
+    }
+    tasks.create<T>("packLlvmFor$hostName") {
+        dependsOn(downloadTaskName)
+        additionalConfiguration()
+        description = "Packs LLVM dependency for $hostName into archive"
```
Comment (ilmat192) [inline dependencyPacker/build.gradle.kts]: (minor) You can specify a group to a task to make it visible in `./gradlew tasks`, e.g. `group = "Distribution packing"`

**Pull request merged**